src/server/pfs/fuse/filesystem.go

```diff
@@ -158,53 +184,95 @@ func (d *directory) Mkdir(ctx context.Context, request *fuse.MkdirRequest) (resu
 
 func (d *directory) Remove(ctx context.Context, req *fuse.RemoveRequest) (retErr error) {
 	defer func() {
-		protolion.Debug(&FileRemove{&d.Node, errorToString(retErr)})
+		if retErr == nil {
+			protolion.Debug(&FileRemove{&d.Node, req.Name, req.Dir, errorToString(retErr)})
+		} else {
+			protolion.Error(&FileRemove{&d.Node, req.Name, req.Dir, errorToString(retErr)})
+		}
 	}()
-	return d.fs.apiClient.DeleteFile(d.Node.File.Commit.Repo.Name, d.Node.File.Commit.ID, filepath.Join(d.Node.File.Path, req.Name))
+	return d.fs.apiClient.DeleteFile(d.Node.File.Commit.Repo.Name,
+		d.Node.File.Commit.ID, filepath.Join(d.Node.File.Path, req.Name), true, d.fs.handleID)
 }
 
 type file struct {
 	directory
 	size    int64
-	local   bool
 	handles []*handle
 }
 
 func (f *file) Attr(ctx context.Context, a *fuse.Attr) (retErr error) {
 	defer func() {
-		protolion.Debug(&FileAttr{&f.Node, &Attr{uint32(a.Mode)}, errorToString(retErr)})
+		if retErr == nil {
+			protolion.Debug(&FileAttr{&f.Node, &Attr{uint32(a.Mode)}, errorToString(retErr)})
+		} else {
+			protolion.Error(&FileAttr{&f.Node, &Attr{uint32(a.Mode)}, errorToString(retErr)})
+		}
 	}()
-	if f.directory.Write {
-		// If the file is from an open commit, we just pretend that it's
-		// an empty file.
-		a.Size = 0
-	} else {
-		fileInfo, err := f.fs.apiClient.InspectFile(
-			f.File.Commit.Repo.Name,
-			f.File.Commit.ID,
-			f.File.Path,
-			f.fs.getFromCommitID(f.getRepoOrAliasName()),
-			f.Shard,
-		)
-		if err != nil && !f.local {
+	fileInfo, err := f.fs.apiClient.InspectFileUnsafe(
+		f.File.Commit.Repo.Name,
+		f.File.Commit.ID,
+		f.File.Path,
+		f.fs.getFromCommitID(f.getRepoOrAliasName()),
+		f.Shard,
+		f.fs.handleID,
+	)
+	if err != nil {
+		return err
+	}
+	if fileInfo != nil {
+		a.Size = fileInfo.SizeBytes
+		a.Mtime = prototime.TimestampToTime(fileInfo.Modified)
+	}
+	a.Mode = 0666
+	a.Inode = f.fs.inode(f.File)
+	return nil
+}
+
+func (f *file) Setattr(ctx context.Context, req *fuse.SetattrRequest, resp *fuse.SetattrResponse) (retErr error) {
+	defer func() {
+		if retErr == nil {
+			protolion.Debug(&FileSetAttr{&f.Node, errorToString(retErr)})
+		} else {
+			protolion.Error(&FileSetAttr{&f.Node, errorToString(retErr)})
+		}
+	}()
+	if req.Size == 0 {
+		err := f.fs.apiClient.DeleteFile(f.Node.File.Commit.Repo.Name,
+			f.Node.File.Commit.ID, f.Node.File.Path, true, f.fs.handleID)
+		if err != nil {
+			return err
+		}
+		if err := f.touch(); err != nil {
```

Comment:
I'm trying to remember the case why we needed to touch the file, and cant. If you can, seems like a good place for a comment